index.js

```diff
@@ -5,15 +5,35 @@ const through = require('through2');
 const debug = require('debug')('tcp-proxy');
 const EventEmitter = require('events').EventEmitter;
 
-function genThrough(interceptor) {
+function genThrough(interceptor, connection) {
+
   return through.obj(function(chunk, enc, done) {
-    const result = interceptor(chunk, enc);
+
+    const context = {
+      client: {
+        ip: connection.client.address().address,
+        port: connection.client.address().port,
+      },
+      server: {
+        ip: connection.server.address().address,
+        port: connection.server.address().port,
+      },
+      self: {
+        ip: connection.forwardHost,
+        port: connection.forwardPort,
+      },
+      size: chunk.length,
+    };
+
+    const result = interceptor(chunk, enc, context);
+
     const handle = data => {
+
       if (data && !Buffer.isBuffer(data)) {
         data = Buffer.from(data);
       }
 
-      done(null, data || chunk);
+      done(null, data || chunk, context);
```

Comment:
This method are no need to pass through `context`. see https://github.com/rvagg/through2

Reply:
Thanks for pointing this out, will be removed.